core/node/consensus/src/storage/connection.rs

```diff
@@ -47,6 +47,28 @@ impl ConnectionPool {
             ctx.sleep(POLL_INTERVAL).await?;
         }
     }
+
+    /// Waits for the `number` L1 batch hash.
+    pub async fn wait_for_batch_hash(
```

Comment:
I reckon this is a method that many would like to see instrumented, to learn how long it takes to wait for the hash.

Reply:
done